### ISSUE TITLE
implement json.set command

### ISSUE THREAD

**Issue by romange** [dragonflydb/dragonfly#238]
See https://docs.aws.amazon.com/AmazonElastiCache/latest/red-ug/json-set.html

Furthermore, when using elasticache redis instance:
```
json.set foo $ '{"a":[], "b":[1], "c":[1,2], "d":[1,2,3]}'
type foo
```

we get `ReJSON-RL`
similarly, `scan 0 type ReJSON-RL` should return `foo`.

Related to #104

**Comment by iko1** [comment]
can you assign it to me please?

**Comment by romange** [comment]
the first step (PR) will be - to support json object inside CompactObject as a separate tag (JSON_TAG part of TagEnum).

**Comment by iko1** [comment]
step zero is merged into the main branch - JSON.SET can set valid data on top of string storage.
I think the command should work on string storage when the existing data within the string is valid JSON and of course on the new JSON type that should be implemented in step one.

**Comment by boazsade** [comment]
The next step should include:
New type entry for "CompactObj" inside "union U" (line 332).
The new type will be dispatched based on a new entry in TagEnum. This means that this new type cannot be a string type. 
This object will help not running the json parsing again and again and it must be aligned as size with "RobjWrapper". 
When a JSON command is dispatched to process value or setting new value, it would lookup or setup the new type that this would create.

**Comment by boazsade** [comment]
This new entry to compact object - the json type, is not compliant with the format the Redis are saving for JSON (the Redis github for JSON support [is here](https://github.com/RedisJSON/RedisJSON), but we are not going to reference it!).

**Comment by boazsade** [comment]
This is over for now, we have an internal issue opened on the missing custom allocator that need to be added to jsoncons 
for this we have a special issue [to update the code once the jsoncons is ready](https://github.com/dragonflydb/.github-private/issues/19). But since the issue [for supporting custome allocator](https://github.com/danielaparker/jsoncons/issues/407) is still open, there is no reason to leave this open as well.
